File: oop.js
// Programacion Orientada a Objetos (OOP)
// Herencia, Encapsulamiento, Polimorfismo y Abstracion

class Persona {
    nombre
    apellido
    genero
    direccion
    edad

    constructor(nombre, apellido, genero, direccion, edad){
        this.nombre = nombre;
        this.apellido = apellido;
        this.genero = genero;
        this.direccion = direccion;
        this.edad = edad;
    }

    comer = function () {

    }
    dormir = function () {
        return "Hora de Dormir";
    }
    correr = function () {

    }
    caminar = function () {

    }
}

let persona = new Persona("Luis", "Rodriguez", "Masculino", "Av Vitacura", 39);

class Estudiante extends Persona {
    constructor(nombre, apellido, genero, direccion, edad){
        super(nombre, apellido, genero, direccion, edad);
    }

    dormir = function(){
        return "Me voy a dormir a las 11";
    }
}

let estudiante = new Estudiante("Luis", "Rodriguez", "Masculino", "Av Vitacura", 39);

console.log(estudiante)

File: oop.py
# Programacion Orientada a Objeto
# Herencia, Encapsulamiento, Polimorfismo y Abstracion

class Persona: 
    nombre = ""
    apellido = ""
    genero = ""
    direccion = ""
    edad = ""

    def __init__(self, nombre, apellido, genero, direccion, edad):
        self.nombre = nombre
        self.apellido = apellido
        self.genero = genero
        self.direccion = direccion
        self.edad = edad

    def comer():
        pass

    def comer():
        pass

    def comer():
        pass

    def comer():
        pass
    
persona = Persona("Luis", "Rodriguez", "Masculino", "Av Vitacura", 39)

print(persona.nombre)

class Estudiante(Persona):
    def __init__(self, nombre, apellido, genero, direccion, edad):
        super().__init__(nombre, apellido, genero, direccion, edad)

    def comer(self):
        return "Hora de comer"

    def __repr__(self):
        return "<Estudiante %s %s>" % (self.nombre, self.apellido)



estudiante = Estudiante("Luis", "Rodriguez", "Masculino", "Av Vitacura", 39)

print(estudiante.__str__)
